internal/services/v1/schema.go

```diff
@@ -46,15 +45,21 @@ type schemaServer struct {
 }
 
 func (ss *schemaServer) ReadSchema(ctx context.Context, in *v1.ReadSchemaRequest) (*v1.ReadSchemaResponse, error) {
-	readRevision, _ := consistency.MustRevisionFromContext(ctx)
-	ds := datastoremw.MustFromContext(ctx).SnapshotReader(readRevision)
+	// Schema is already read from the head revision.
+	ds := datastoremw.MustFromContext(ctx)
+	headRevision, err := ds.HeadRevision(ctx)
+	if err != nil {
+		return nil, rewriteError(ctx, err)
+	}
+
+	reader := ds.SnapshotReader(headRevision)
```

Comment:
a potential optimization would be to have a method like `HeadReader` which executes the first query and returning the revision, and uses that revision for any potential subsequent queries. That way we avoid one roundtrip

Reply:
Not sure how it avoids a roundtrip; `SnapshotReader(...)` should not be making a DB call until its actually used